### ISSUE TITLE
If you use a large video or video list as the source, it suddenly becomes slow as shown below.

### ISSUE THREAD

**Issue by RohTaeHyeon** [ultralytics/ultralytics#17125]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

If you use a large video or video list as the source, it suddenly becomes slow as shown below. Can you confirm?

my code just...
## Load custum yolov8 training result weight pt file ##

from ultralytics import YOLO
model = YOLO('/data1/YOLO/ultralytics/runs/detect/train5/weights/best.pt') 
results = model(source='rtsp148/rtsp148_output_20241007_113911.mp4',conf=0.5,save=True)

video 1/1 (frame 7345/9007) rtsp148/rtsp148_output_20241007_113911.mp4: 384x640 1 person, 1 head, 4.7ms
video 1/1 (frame 7346/9007) rtsp148/rtsp148_output_20241007_113911.mp4: 384x640 1 person, 1 head, 4.4ms
video 1/1 (frame 7347/9007) rtsp148/rtsp148_output_20241007_113911.mp4: 384x640 1 person, 1 head, 4.8ms
video 1/1 (frame 7348/9007) rtsp148/rtsp148_output_20241007_113911.mp4: 384x640 1 person, 1 head, 5.3ms
video 1/1 (frame 7349/9007) rtsp148/rtsp148_output_20241007_113911.mp4: 384x640 1 person, 1 head, 4.5ms
video 1/1 (frame 7350/9007) rtsp148/rtsp148_output_20241007_113911.mp4: 384x640 1 person, 1 head, 4.4ms
video 1/1 (frame 7351/9007) rtsp148/rtsp148_output_20241007_113911.mp4: 384x640 1 person, 1 head, 4.8ms
video 1/1 (frame 7352/9007) rtsp148/rtsp148_output_20241007_113911.mp4: 384x640 1 person, 1 head, 4.4ms
video 1/1 (frame 7353/9007) rtsp148/rtsp148_output_20241007_113911.mp4: 384x640 1 person, 1 head, 4.4ms
video 1/1 (frame 7354/9007) rtsp148/rtsp148_output_20241007_113911.mp4: 384x640 1 person, 1 head, 4.5ms
video 1/1 (frame 7355/9007) rtsp148/rtsp148_output_20241007_113911.mp4: 384x640 1 person, 1 head, 4.6ms
video 1/1 (frame 7356/9007) rtsp148/rtsp148_output_20241007_113911.mp4: 384x640 1 person, 1 head, 4.4ms
video 1/1 (frame 7357/9007) rtsp148/rtsp148_output_20241007_113911.mp4: 384x640 1 person, 1 head, 5.7ms
video 1/1 (frame 7358/9007) rtsp148/rtsp148_output_20241007_113911.mp4: 384x640 1 person, 1 head, 43.0ms
video 1/1 (frame 7359/9007) rtsp148/rtsp148_output_20241007_113911.mp4: 384x640 1 person, 1 head, 104.0ms
video 1/1 (frame 7360/9007) rtsp148/rtsp148_output_20241007_113911.mp4: 384x640 1 person, 1 head, 98.4ms
video 1/1 (frame 7361/9007) rtsp148/rtsp148_output_20241007_113911.mp4: 384x640 1 person, 1 head, 2412.1ms
video 1/1 (frame 7362/9007) rtsp148/rtsp148_output_20241007_113911.mp4: 384x640 1 person, 1 head, 189.8ms

my env : conda python=3.8
_libgcc_mutex             0.1                        main  
_openmp_mutex             5.1                       1_gnu  
aiohappyeyeballs          2.4.0                    pypi_0    pypi
aiohttp                   3.10.5                   pypi_0    pypi
aiosignal                 1.3.1                    pypi_0    pypi
async-timeout             4.0.3                    pypi_0    pypi
attrs                     24.2.0                   pypi_0    pypi
blas                      1.0                         mkl  
brotli-python             1.0.9            py38h6a678d5_8  
bzip2                     1.0.8                h5eee18b_6  
ca-certificates           2024.7.2             h06a4308_0  
certifi                   2024.8.30        py38h06a4308_0  
charset-normalizer        3.3.2              pyhd3eb1b0_0  
contourpy                 1.1.1                    pypi_0    pypi
cuda-cudart               12.1.105                      0    nvidia
cuda-cupti                12.1.105                      0    nvidia
cuda-libraries            12.1.0                        0    nvidia
cuda-nvrtc                12.1.105                      0    nvidia
cuda-nvtx                 12.1.105                      0    nvidia
cuda-opencl               12.6.68                       0    nvidia
cuda-python               12.3.0                   pypi_0    pypi
cuda-runtime              12.1.0                        0    nvidia
cuda-version              12.6                          3    nvidia
cycler                    0.12.1                   pypi_0    pypi
ffmpeg                    4.3                  hf484d3e_0    pytorch
filelock                  3.13.1           py38h06a4308_0  
fonttools                 4.53.1                   pypi_0    pypi
freetype                  2.12.1               h4a9f257_0  
frozenlist                1.4.1                    pypi_0    pypi
fsspec                    2024.9.0                 pypi_0    pypi
gevent                    24.2.1                   pypi_0    pypi
geventhttpclient          2.0.2                    pypi_0    pypi
gmp                       6.2.1                h295c915_3  
gmpy2                     2.1.2            py38heeb90bb_0  
gnutls                    3.6.15               he1e5248_0  
greenlet                  3.0.3                    pypi_0    pypi
grpcio                    1.66.1                   pypi_0    pypi
idna                      3.7              py38h06a4308_0  
importlib-resources       6.4.5                    pypi_0    pypi
intel-openmp              2023.1.0         hdb19cb5_46306  
jinja2                    3.1.4            py38h06a4308_0  
jpeg                      9e                   h5eee18b_3  
kiwisolver                1.4.7                    pypi_0    pypi
lame                      3.100                h7b6447c_0  
lcms2                     2.12                 h3be6417_0  
ld_impl_linux-64          2.38                 h1181459_1  
lerc                      3.0                  h295c915_0  
libcublas                 12.1.0.26                     0    nvidia
libcufft                  11.0.2.4                      0    nvidia
libcufile                 1.11.1.6                      0    nvidia
libcurand                 10.3.7.68                     0    nvidia
libcusolver               11.4.4.55                     0    nvidia
libcusparse               12.0.2.55                     0    nvidia
libdeflate                1.17                 h5eee18b_1  
libffi                    3.4.4                h6a678d5_1  
libgcc-ng                 11.2.0               h1234567_1  
libgomp                   11.2.0               h1234567_1  
libiconv                  1.16                 h5eee18b_3  
libidn2                   2.3.4                h5eee18b_0  
libjpeg-turbo             2.0.0                h9bf148f_0    pytorch
libnpp                    12.0.2.50                     0    nvidia
libnvjitlink              12.1.105                      0    nvidia
libnvjpeg                 12.1.1.14                     0    nvidia
libpng                    1.6.39               h5eee18b_0  
libstdcxx-ng              11.2.0               h1234567_1  
libtasn1                  4.19.0               h5eee18b_0  
libtiff                   4.5.1                h6a678d5_0  
libunistring              0.9.10               h27cfd23_0  
libwebp-base              1.3.2                h5eee18b_0  
llvm-openmp               14.0.6               h9e868ea_0  
lz4-c                     1.9.4                h6a678d5_1  
markupsafe                2.1.3            py38h5eee18b_0  
matplotlib                3.7.5                    pypi_0    pypi
mkl                       2023.1.0         h213fc3f_46344  
mkl-service               2.4.0            py38h5eee18b_1  
mkl_fft                   1.3.8            py38h5eee18b_0  
mkl_random                1.2.4            py38hdb19cb5_0  
mpc                       1.1.0                h10f8cd9_1  
mpfr                      4.0.2                hb69a4c5_1  
mpmath                    1.3.0            py38h06a4308_0  
multidict                 6.1.0                    pypi_0    pypi
ncurses                   6.4                  h6a678d5_0  
nettle                    3.7.3                hbbd107a_1  
networkx                  3.1              py38h06a4308_0  
numpy                     1.24.3           py38hf6e8229_1  
numpy-base                1.24.3           py38h060ed82_1  
nvidia-cuda-runtime-cu12  12.6.68                  pypi_0    pypi
onnx                      1.16.2                   pypi_0    pypi
opencv-python             4.10.0.84                pypi_0    pypi
openh264                  2.1.1                h4ff587b_0  
openjpeg                  2.5.2                he7f1fd0_0  
openssl                   3.0.15               h5eee18b_0  
packaging                 24.1                     pypi_0    pypi
pandas                    2.0.3                    pypi_0    pypi
pillow                    10.4.0           py38h5eee18b_0  
pip                       24.2             py38h06a4308_0  
protobuf                  4.25.4                   pypi_0    pypi
psutil                    6.0.0                    pypi_0    pypi
py-cpuinfo                9.0.0                    pypi_0    pypi
pyparsing                 3.1.4                    pypi_0    pypi
pysocks                   1.7.1            py38h06a4308_0  
python                    3.8.19               h955ad1f_0  
python-dateutil           2.9.0.post0              pypi_0    pypi
python-rapidjson          1.20                     pypi_0    pypi
pytorch                   2.4.1           py3.8_cuda12.1_cudnn9.1.0_0    pytorch
pytorch-cuda              12.1                 ha16c6d3_5    pytorch
pytorch-mutex             1.0                        cuda    pytorch
pytz                      2024.1                   pypi_0    pypi
pyyaml                    6.0.1            py38h5eee18b_0  
readline                  8.2                  h5eee18b_0  
requests                  2.32.3           py38h06a4308_0  
scipy                     1.10.1                   pypi_0    pypi
seaborn                   0.13.2                   pypi_0    pypi
setuptools                72.1.0           py38h06a4308_0  
shapely                   2.0.6                    pypi_0    pypi
six                       1.16.0                   pypi_0    pypi
sqlite                    3.45.3               h5eee18b_0  
sympy                     1.13.2           py38h06a4308_0  
tbb                       2021.8.0             hdb19cb5_0  
tensorrt                  10.1.0                   pypi_0    pypi
tensorrt-cu12             10.4.0                   pypi_0    pypi
tensorrt-cu12-bindings    10.4.0                   pypi_0    pypi
tensorrt-cu12-libs        10.4.0                   pypi_0    pypi
tk                        8.6.14               h39e8969_0  
torchaudio                2.4.1                py38_cu121    pytorch
torchtriton               3.0.0                      py38    pytorch
torchvision               0.19.1               py38_cu121    pytorch
tqdm                      4.66.5                   pypi_0    pypi
tritonclient              2.49.0                   pypi_0    pypi
typing_extensions         4.11.0           py38h06a4308_0  
tzdata                    2024.1                   pypi_0    pypi
ultralytics               8.2.91                   pypi_0    pypi
ultralytics-thop          2.0.6                    pypi_0    pypi
urllib3                   2.2.2            py38h06a4308_0  
wheel                     0.43.0           py38h06a4308_0  
xz                        5.4.6                h5eee18b_1  
yaml                      0.2.5                h7b6447c_0  
yarl                      1.11.1                   pypi_0    pypi

**Comment by UltralyticsAssistant** [comment]
👋 Hello @RohTaeHyeon, thank you for reaching out and for your interest in Ultralytics 🚀! This is an automated response to guide you through initial troubleshooting steps, and an Ultralytics engineer will also assist you soon.

We recommend checking out our [Docs](https://docs.ultralytics.com) where you may find solutions to common questions and issues. You can explore various [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) examples that might be helpful.

For your issue on video processing becoming unexpectedly slow, if this is a 🐛 Bug Report, please try to provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) to help us better understand and address the problem.

To ensure you're using the latest features and fixes, please upgrade your `ultralytics` package and check that your environment is properly set up. You can do this by running the following command:

```bash
pip install -U ultralytics
```

Make sure all dependencies, including [CUDA](https://developer.nvidia.com/cuda) and [PyTorch](https://pytorch.org/get-started/locally/), are up to date.

For community support and discussions, consider joining our [Discord](https://ultralytics.com/discord) 🎧 for real-time chat, participate in [Discourse](https://community.ultralytics.com) for in-depth discussions, or engage with our community on [Reddit](https://reddit.com/r/ultralytics).

## Verified Environments

The following environments are verified to ensure a good experience with YOLO:

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud**, **Amazon**, and **Docker** environments are also available with guidance in our docs for setting them up.

## CI Status

Check the latest Ultralytics CI status here: 

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all tests are passing, confirming the integrity of YOLO's functionality across various platforms.

We're here to help, and will get back to you as soon as possible with further assistance!

**Comment by Y-T-G** [comment]
You should use `stream=True` if you are using a large video.

https://docs.ultralytics.com/modes/predict/#key-features-of-predict-mode

**Comment by WandererGuy** [comment]
for anyone in future interest on no how to use stream, this is how, 
    for single_frame_result in model.predict(
    source = folder_frame,
    batch=8,
    conf=MIN_CONF,
    imgsz=640,
    verbose=False,
    save=False,
    stream=True
):  
        count += 1
        logging.info ('prediction frame: ' + str(count))
        for item in single_frame_result:
                ori_img = item.orig_img
                box = item.boxes
